src/update.ts

```diff
@@ -345,25 +345,27 @@ export const makeUpdate =
             }
           : never,
       ): UpdateQuery<T, number> {
-        const keys = Object.keys(values);
+        const valuesToken = [];
+        for (const key of Object.keys(values)) {
+          const value = (values as any)[key];
+          if (value === undefined) continue;
+
+          const column = (table as any)[key] as Column<any, any, any, any, any, any>;
+
+          valuesToken.push(
+            new CollectionToken([
+              new StringToken(wrapQuotes(column.getSnakeCaseName())),
+              new StringToken(`=`),
+              ...(isTokenable(value) ? value.toTokens() : [new ParameterToken(value)]),
+            ]),
+          );
+        }
 
         return new UpdateQuery(queryExecutor, [], table, 'AFFECTED_COUNT', [
           new StringToken(`UPDATE`),
           ...table.toTokens(),
           new StringToken(`SET`),
-          new SeparatorToken(
-            `,`,
-            keys.map((key) => {
-              const column = (table as any)[key] as Column<any, any, any, any, any, any>;
-              const value = (values as any)[key];
-
-              return new CollectionToken([
-                new StringToken(wrapQuotes(column.getSnakeCaseName())),
-                new StringToken(`=`),
-                ...(isTokenable(value) ? value.toTokens() : [new ParameterToken(value)]),
```

Comment:
Hmm, where `undefined` turning into a DB `null`? Was that in `ParameterToken`? If so, maybe worth adding an `assert` so it can't inadvertently happen anymore?

Reply:
We could. However, there are potential where it might be desirable since ParameterToken has many entry points. 

We could temporarily turn it into a log message for it so that we can see if it ever reaches it in practice.

Reply:
> However, there are potential where it might be desirable since ParameterToken has many entry points.

That's actually what I was worried about -- other callers that are using `undefined` to mean `null`, which might result in subtle bugs.

Logging is a good option.

But I'm still curious _where_ the JS `undefined` is being turned into a DB `NULL`. Is it being done by Mammoth? Or by the `pg` library?

Reply:
pg is doing it. If you remove the undefined check in this code and run the test, you can see the parameters are still sent as undefined.

Reply:
Array undefined -> null: https://github.com/brianc/node-postgres/blob/92cb640fd316972e323ced6256b2acd89b1b58e0/packages/pg/lib/utils.js#L21
prepareValue: https://github.com/brianc/node-postgres/blob/92cb640fd316972e323ced6256b2acd89b1b58e0/packages/pg/lib/utils.js#L48

Reply:
(So if we were to log for undefined values, we should log in our code base in DbClientImpl)

Reply:
Oh nice find. So let's add logging in our app to make sure we're not relying on `undefined`. And then we can add the assert to Mammoth.